ibis/expr/operations.py

```diff
@@ -804,6 +872,38 @@ class Variance(VarianceBase):
     pass
 
 
+class Correlation(Reduction):
+    """
+    coefficient of correlation of a set of number pairs.
+
+    """
+    left = Arg(rlz.numeric)
+    right = Arg(rlz.numeric)
+    how = Arg(rlz.isin({'sample', 'pop'}), default=None)
+    where = Arg(rlz.boolean, default=None)
+
+    def output_type(self):
+        if isinstance(self.left, ir.DecimalValue):
+            dtype = self.left.type().largest
+        else:
+            dtype = dt.float64
+        return dtype.scalar_type()
```

Comment:
@cpcloud  I'm not sure about the type promotion here, do We want decimals here?

Reply:
There is also [rules.numeric_like](https://github.com/ibis-project/ibis/blob/master/ibis/expr/rules.py#L370) which we might want to extend.

Reply:
@xmnlab You can leave out decimal types for now and if/when we need support for them we can add it. We can also add it in a follow up PR before that as well.

Reply:
great! so should I keep that, right?

Reply:
No, leave them out for now